### PR TITLE
[IMP] im_livechat: speed up operator selection

### PR DESCRIPTION
**Pull request by odony**
Since #135113, calls to /im_livechat/get_session can become very expensive on large databases with a long history of livechat sessions. Operators who have thousands of past livechat sessions (channels) can cause the SQL query in get_operator() to take multiple seconds.

When the livechat is set to auto-popup and there are lots of visitors, this can become very significant.

This patch changes two aspects:
- introduce a CTE for the RTC session part, in order to avoid a JOIN cardinality explosion between livechat channels and channel_members - we only care about channel membership for a RTC session AFAICT
- restrict the selection of livechat channels created **within the last 24h**, considering that we only care about messages sent in the last 30 minutes anyway

The second part may change the result ordering in the presence of old channels (>24h) with recent messages - but that is rather unlikely, and an operator will be selected anyway.

This change makes a huge difference on a database with 300k livechat channels and 1 million `discuss_channel_member` records, from several seconds to >50ms. The results are identical or extremely similar in most cases, and the database only needs to look at a few hundred records instead of millions.

### CODE REVIEW

File: addons/im_livechat/models/im_livechat_channel.py [hunk 1]
```diff
@@ -228,12 +228,16 @@ def _get_operator(self, previous_operator_id=None, lang=None, country_id=None):
         if not self.available_operator_ids:
             return False
         self.env.cr.execute("""
+            WITH operator_rtc_session AS (
+                SELECT s.id, member.partner_id as partner_id
+                  FROM discuss_channel_rtc_session s
+                  JOIN discuss_channel_member member ON (member.id = s.channel_member_id)
```
Comment (nseinlet) [inline addons/im_livechat/models/im_livechat_channel.py:234]:
```suggestion
                SELECT COUNT(DISTINCT s.id) as nbr, member.partner_id as partner_id
                  FROM discuss_channel_rtc_session s
                  JOIN discuss_channel_member member ON (member.id = s.channel_member_id)
                  GROUP BY member.partner_id
```
and update subsequent query?

**Review by tsm-odoo — APPROVED** [review]
LGTM even if results were to be slightly different it would not matter. Thanks for the fix!

**Comment by tsm-odoo** [comment]
:arrow_up: updated to apply @nseinlet suggestion

**Comment by tsm-odoo** [comment]
@odony I applied your suggestion, I think this PR can be merged now.

**Pull request closed without merging**